util/helm/creds.go

```diff
@@ -146,11 +147,37 @@ func (creds AzureWorkloadIdentityCreds) GetAccessToken() (string, error) {
 		return "", fmt.Errorf("failed to get Azure access token after challenge: %w", err)
 	}
 
-	// Access token has a lifetime of 3 hours
-	storeAzureToken(key, token, 2*time.Hour)
+	tokenExpiry, err := getJWTExpiry(token)
+	if err != nil {
+		tokenExpiry = time.Now().Add(15 * time.Minute)
+	}
+	// Azure Container Registry does not provide expires_on in the response
+	// so we set a default expiration time of 10 minutes
+	storeAzureToken(key, token, tokenExpiry.Sub(time.Now().Add(5*time.Minute)))
 	return token, nil
 }
 
+func getJWTExpiry(token string) (time.Time, error) {
```

Comment:
I'd look into using the jwt library: https://pkg.go.dev/github.com/golang-jwt/jwt/v4#Parser

Reply:
Thanks I will use this library.